sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.netty;
+
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.util.logging.ClientLogger;
+import io.netty.channel.nio.NioEventLoopGroup;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.tcp.ProxyProvider;
+
+/**
+ *
+ */
+public class NettyAsyncHttpClientBuilder {
+    private final ClientLogger logger = new ClientLogger(NettyAsyncHttpClientBuilder.class);
+
+    private ProxyOptions proxyOptions;
+    private boolean enableWiretap;
+    private int port = 80;
+    private NioEventLoopGroup nioEventLoopGroup;
+
+    /**
+     *
+     */
+    public NettyAsyncHttpClientBuilder() {
+    }
+
+    /**
+     *
+     * @return A new NettyAsyncHttpClient instance
+     * @throws IllegalStateException If proxy type is unknown.
+     */
+    public NettyAsyncHttpClient build() {
+        HttpClient nettyHttpClient = HttpClient.create()
+            .port(port)
+            .wiretap(enableWiretap)
+            .tcpConfiguration(tcpConfig -> {
+                if (nioEventLoopGroup != null) {
+                    tcpConfig = tcpConfig.runOn(nioEventLoopGroup);
+                }
+
+                if (proxyOptions == null) {
+                    return tcpConfig;
+                }
+
+                ProxyProvider.Proxy nettyProxy;
+                switch (proxyOptions.type()) {
+                    case HTTP: nettyProxy = ProxyProvider.Proxy.HTTP; break;
+                    case SOCKS4: nettyProxy = ProxyProvider.Proxy.SOCKS4; break;
+                    case SOCKS5: nettyProxy = ProxyProvider.Proxy.SOCKS5; break;
+                    default:
+                        throw logger.logExceptionAsWarning(new IllegalStateException("Unknown Proxy type '" + proxyOptions.type() + "' in use. Not configuring Netty proxy."));
+                }
+
+                return tcpConfig.proxy(ts -> ts.type(nettyProxy).address(proxyOptions.address()));
```

Comment:
To avoid a future bug, we should move the configuration of the proxy into the if block above by inverting the null check.